classes/local/utils.php

```diff
@@ -0,0 +1,482 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tiny_elements\local;
+
+/**
+ * Utility class for tiny_elements.
+ *
+ * @package    tiny_elements
+ * @copyright  2024 ISB Bayern
+ * @author     Philipp Memmel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class utils {
+    /** @var array $flavors */
+    private static array $flavors;
+
+    /**
+     * Get all components.
+     * @param bool $isstudent
+     * @return array all components
+     */
+    public static function get_all_components(bool $isstudent = false): array {
```

Comment:
`$isstudent` is not used -> remove from signature?

Also check the other methods in this class regarding `$isstudent`

Reply:
It is used in line 36 - removed it from other methods where it is not used (yet).